test/tests/functional/pbs_maintenance_reservations.py

```diff
@@ -694,3 +694,25 @@ def test_maintenance_degrade_reservation_jobs_dont_run(self):
         a = {'comment': 'Can Never Run: Reservation is in an invalid state',
              'job_state': 'Q'}
         self.server.expect(JOB, a, id=jid)
+
+    def test_maintenance_parse_numerous_hosts(self):
+        """
+        Test if the parsing of numerous hosts succeed.
+        """
+        now = int(time.time())
+
+        a = {'reserve_start': now + 3600,
+             'reserve_end': now + 7200}
+        chars = 'abcdefghijklmnopqrstuvwxyz'
+        h = [''.join(random.choices(chars, k=8)) for i in range(200)]
+        r = Reservation(TEST_USER, attrs=a, hosts=h)
+
+        msg = ""
+
+        try:
```

Comment:
If you are expecting an exception, please consider using "with assertRaises(PbsSubmitError) as" syntax.

Reply:
Thank you @arungrover I applied your suggestion for the whole test suite.